### PR TITLE
update module using module_utils/cloudfront_facts.py

### PR DESCRIPTION
**Pull request by abikouo**
Depends-On: https://github.com/ansible-collections/amazon.aws/pull/1265
##### SUMMARY

update cloudfront_* modules, fix some bugs and add integration tests

##### ISSUE TYPE

- Feature Pull Request

### CODE REVIEW

**Comment by jillr** [comment]
When running the cloudfront_distribution tests to test the terminator PR, they pass but there are 2 distributions remaining in the CI AWS account that are not automatically deleted by the test suite.  Please ensure that all resources created by the tests are removed by the tests.

**Comment by abikouo** [comment]
> When running the cloudfront_distribution tests to test the terminator PR, they pass but there are 2 distributions remaining in the CI AWS account that are not automatically deleted by the test suite. Please ensure that all resources created by the tests are removed by the tests.

only the second cloudfront distribution was not deleted, I have updated the test. Could you please validate once again in a clean environment? Thanks

**Comment by gravesm** [comment]
recheck

**Comment by gravesm** [comment]
```
2022-12-01 16:41:30.567401 | controller | TASK [cloudfront_distribution : check that custom origin uses the provided origin_access_identity] ***
2022-12-01 16:41:30.567408 | controller | task path: /home/zuul/.ansible/collections/ansible_collections/community/aws/tests/integration/targets/cloudfront_distribution/tasks/main.yml:505
2022-12-01 16:41:30.643578 | controller | fatal: [testhost]: FAILED! => {
2022-12-01 16:41:30.643613 | controller |     "assertion": "update_distribution_with_specific_access_identity.origins['items'].0.s3_origin_config.origin_access_identity == origin_access_identity",
```
I saw this failure locally, too, but when I re-ran all the tests passed. Something here seems a little flaky.

File: tests/integration/targets/cloudfront_distribution/tasks/main.yml [hunk 1]
```diff
@@ -86,30 +86,30 @@
...
-      distribution_id: "{{ distribution_id }}"
-      purge_origins: no
-      state: present
-    register: cf_dist_with_id
-
-  - name: ensure distribution was not updated
-    assert:
-      that:
-        - not cf_dist_with_id.changed
+  # - name: re-run cloudfront distribution with same defaults
```
[the remark this answers is not in the document]
Comment (alinabuzachis) [reply tests/integration/targets/cloudfront_distribution/tasks/main.yml:89]: Ok, can you please open an issue for idempotency?

**Pull request merged**